recirq/quantum_chess/dynamic_look_ahead_heuristic_circuit_transformer_test.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2020 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+import pytest
+from collections import deque
+
+import cirq
+
+import recirq.quantum_chess.circuit_transformer as ct
+
+a0 = cirq.NamedQubit('a0')
+a1 = cirq.NamedQubit('a1')
+a2 = cirq.NamedQubit('a2')
+a3 = cirq.NamedQubit('a3')
+a4 = cirq.NamedQubit('a4')
+a5 = cirq.NamedQubit('a5')
+a6 = cirq.NamedQubit('a6')
+a7 = cirq.NamedQubit('a7')
+
+logical = {
+    a0: [(a2, 0), (a1, 1), (a3, 6)],
+    a1: [(a0, 1), (a2, 3)],
+    a2: [(a0, 0), (a1, 3), (a3, 5)],
+    a3: [(a2, 5), (a0, 6)],
+}
+
+logical2 = {
+    a0: [(a2, 0), (a1, 1), (a3, 6)],
+    a1: [(a0, 1), (a2, 3), (a7, math.inf)],
+    a2: [(a0, 0), (a1, 3), (a3, 5)],
+    a3: [(a2, 5), (a0, 6)],
+    a4: [(a5, 0), (a6, math.inf)],
+    a5: [(a4, 0)],
+    a6: [(a7, 0), (a4, math.inf)],
+    a7: [(a6, 0), (a1, math.inf)],
+}
+
+zero_zero = cirq.GridQubit(0, 0)
+zero_one = cirq.GridQubit(0, 1)
+zero_two = cirq.GridQubit(0, 2)
+zero_three = cirq.GridQubit(0, 3)
+zero_four = cirq.GridQubit(0, 4)
+zero_five = cirq.GridQubit(0, 5)
+zero_six = cirq.GridQubit(0, 6)
+zero_seven = cirq.GridQubit(0, 7)
+zero_eight = cirq.GridQubit(0, 8)
+zero_nine = cirq.GridQubit(0, 9)
+zero_ten = cirq.GridQubit(0, 10)
+one_zero = cirq.GridQubit(1, 0)
+one_one = cirq.GridQubit(1, 1)
+one_two = cirq.GridQubit(1, 2)
+one_three = cirq.GridQubit(1, 3)
+one_four = cirq.GridQubit(1, 4)
+one_five = cirq.GridQubit(1, 5)
+one_six = cirq.GridQubit(1, 6)
+one_seven = cirq.GridQubit(1, 7)
+one_eight = cirq.GridQubit(1, 8)
+one_nine = cirq.GridQubit(1, 9)
+one_ten = cirq.GridQubit(1, 10)
```

Comment:
would it be cleaner to store these in an array/dict and construct them in a loop?
ex something like:
```
grid_qubits = dict(
    ('{row}_{col}', cirq.GridQubit(row, col))
    for row in range(2) for col in range(11))
    
... grid_qubits['1_10'] ...
```

Reply:
done!